sql/catalyst/src/main/scala/org/apache/spark/sql/errors/SqlScriptingException.scala

```diff
@@ -17,40 +17,77 @@
 
 package org.apache.spark.sql.errors
 
-import org.apache.spark.SparkException
+import org.apache.spark.{SparkException, SparkThrowableHelper}
+import org.apache.spark.sql.catalyst.trees.Origin
+import org.apache.spark.sql.errors.SqlScriptingException.errorMessageWithLineNumber
+
+class SqlScriptingException protected (
+    origin: Origin,
+    errorClass: String,
+    cause: Throwable,
+    messageParameters: Map[String, String] = Map.empty)
+  extends SparkException(
```

Comment:
This is fine, but I think it can be further simplified, we can extend `Exception(message, cause)` the same way that `SparkException` does it, and implementing a few simple but required override functions. reasoning: there are properties in `SparkException` that we actually don't need.